scripts/confix/confix.go

```diff
@@ -85,21 +85,138 @@ var plan = transform.Plan{
 		T:    transform.SnakeToKebab(),
 	},
 	{
-		// Since https://github.com/tendermint/tendermint/pull/6896.
-		Desc:    "Rename [fastsync] to [blocksync]",
-		T:       transform.Rename(parser.Key{"fastsync"}, parser.Key{"blocksync"}),
+		// [fastsync]  renamed in https://github.com/tendermint/tendermint/pull/6896.
+		// [blocksync] removed in https://github.com/tendermint/tendermint/pull/7159.
+		Desc: "Remove [fastsync] and [blocksync] sections",
+		T: transform.Func(func(_ context.Context, doc *tomledit.Document) error {
+			doc.First("fast-sync").Remove()
+			transform.FindTable(doc, "fastsync").Remove()
+			transform.FindTable(doc, "blocksync").Remove()
+			return nil
+		}),
+		ErrorOK: true,
+	},
+	{
+		// Since https://github.com/tendermint/tendermint/pull/6241.
+		//
+		// TODO(creachadair): backport into v0.35.x.
+		Desc: `Add top-level mode setting (default "full")`,
+		T: transform.EnsureKey(nil, &parser.KeyValue{
+			Block: parser.Comments{"Mode of Node: full | validator | seed"},
+			Name:  parser.Key{"mode"},
+			Value: parser.MustValue(`"full"`),
+		}),
 		ErrorOK: true,
 	},
 	{
-		// Since https://github.com/tendermint/tendermint/pull/7159.
-		Desc: "Move top-level fast_sync key to blocksync.enable",
-		T: transform.MoveKey(
-			parser.Key{"fast-sync"},
-			parser.Key{"blocksync"},
-			parser.Key{"enable"},
+		// Since https://github.com/tendermint/tendermint/pull/7121.
+		Desc: "Remove gRPC settings from the [rpc] section",
+		T: transform.Func(func(_ context.Context, doc *tomledit.Document) error {
+			doc.First("rpc", "grpc-laddr").Remove()
+			doc.First("rpc", "grpc-max-open-connections").Remove()
+			return nil
+		}),
+	},
+	{
+		// Since https://github.com/tendermint/tendermint/pull/8217.
+		Desc: "Remove per-node consensus timeouts (converted to consensus parameters)",
+		T: transform.Remove(
+			parser.Key{"consensus", "skip-timeout-commit"},
+			parser.Key{"consensus", "timeout-commit"},
+			parser.Key{"consensus", "timeout-precommit"},
+			parser.Key{"consensus", "timeout-precommit-delta"},
+			parser.Key{"consensus", "timeout-prevote"},
+			parser.Key{"consensus", "timeout-prevote-delta"},
+			parser.Key{"consensus", "timeout-propose"},
+			parser.Key{"consensus", "timeout-propose-delta"},
```

Comment:
Does this transformation remove the associated comments? Perhaps... should it?

Reply:
It does. Comments are "attached" to the items they're adjacent to, so comments on an item move (or are removed) with it.

Reply:
(if you think we should keep the comments, that is also possible, but will require a bit of extra work)

Reply:
I think it should! That seems very sensible.